tarsier/core.py

```diff
@@ -27,24 +29,48 @@ def __init__(self, ocr_service: OCRService):
 
     async def page_to_image(
         self, driver: AnyDriver, tag_text_elements: bool = False, tagless: bool = False
-    ) -> Tuple[bytes, Dict[int, str]]:
+    ) -> Tuple[bytes, bytes, Dict[int, str]]:
         adapter = adapter_factory(driver)
+        initial_screenshot = await self._take_screenshot(adapter)
         tag_to_xpath = (
             await self._tag_page(adapter, tag_text_elements) if not tagless else {}
         )
-        screenshot = await self._take_screenshot(adapter)
+        await self._hide_non_tag_elem(adapter)
+        tagged_screenshot = await self._take_screenshot(adapter)
+        await self._revert_visibilities(adapter)
         if not tagless:
             await self._remove_tags(adapter)
-        return screenshot, tag_to_xpath if not tagless else {}
+        return (
+            initial_screenshot,
+            tagged_screenshot,
+            tag_to_xpath if not tagless else {},
+        )
```

Comment:
Let's maybe keep this method the same and just move over logic to page_to_text

This is because anyone using `page_to_image` currently expects a single image with all of the tagged elements (Alongside the page text)